ansible_base/rbac/managed.py

```diff
@@ -0,0 +1,145 @@
+import logging
+
+from django.utils.translation import gettext_lazy as _
+from django.utils.translation import gettext_noop
+
+logger = logging.getLogger('ansible_base.rbac.managed')
+
+
+class ManagedRoleDefinition:
+    def __init__(self, apps, overrides=None):
+        if overrides:
+            for key, value in overrides.items():
+                setattr(self, key, value)
+
+    def get_model(self):
+        raise NotImplementedError
+
+    def get_permissions(self) -> set[str]:
+        raise NotImplementedError
+
+    def get_translated_name(self) -> str:
+        return _(self.name)
+
+    def get_content_type(self, apps):
+        content_type_cls = apps.get_model('contenttypes', 'ContentType')
+        return content_type_cls.objects.get_for_model(self.get_model())
+
+    def get_or_create(self, apps):
+        "Create from a list of text-type permissions and do validation"
+        role_definition_cls = apps.get_model('dab_rbac', 'RoleDefinition')
+        defaults = {
+            'description': self.description,
+            'content_type': self.get_content_type(apps),
+            'managed': True,
+        }
+        rd, created = role_definition_cls.objects.get_or_create(name=self.name, defaults=defaults)
+
+        if created:
+            permissions = self.get_permissions()
+            permission_cls = apps.get_model('dab_rbac', 'DABPermission')
+            perm_list = [permission_cls.objects.get(codename=str_perm) for str_perm in permissions]
+            rd.permissions.add(*perm_list)
+            logger.info(f'Created {self.shortname} managed role definition, name={self.name}')
+            logger.debug(f'Data of {self.name} role definition: {defaults}')
+            logger.debug(f'Permissions of {self.name} role definition: {permissions}')
+        return rd, created
+
+
+class AdminMixin:
+    def get_permissions(self) -> set[str]:
+        """All permissions possible for the associated model"""
+        from ansible_base.rbac.validators import combine_values, permissions_allowed_for_role
+
+        return combine_values(permissions_allowed_for_role(self.get_model()))
+
+
+class OrganizationMixin:
+    def get_model(self):
+        from ansible_base.lib.utils.auth import get_organization_model
+
+        return get_organization_model()
```

Comment:
@john-westcott-iv I wanted to use the shared method for doing this, but after getting further in development it's just not going to work. I want to further customize this _in settings_ so no imports or python objects. It makes the most sense to use the setting and have the logic here do `get_model`. Also we can't import from utils at top level. I'll leave a comment about this tension.

Reply:
Yea, I've been thinking about `uitls` lately and wondering if we should take anything with a model import out so that utils would /always/ be safe to import.

Reply:
Documented https://github.com/ansible/django-ansible-base/issues/443